Stellascope Port

XanderDox opened the pull request:
**The Stellascope**

This PR ports a neat little tool from Aurora, changed to be more simple and use existing hologram projector item framework due to not having a specific lore-connection here as it does there.

The Stellascope is a small antique holographic projector that can create a display of a rotating celestial projection. It jingles when it does so.

Projections are limited to one at a time to avoid spam.

Item is available in the loadout under utility.

https://github.com/user-attachments/assets/f082527a-e3cb-4b8d-a7f6-2a6e3a5dfc85
:cl:
rscadd: Adds the Stellascope as a hologram projector item, selectable in loadout under utility.
imageadd: Ports the Stellascope and projection sprites from Aurora.
soundadd: Ports the 'Jingle.ogg' from Aurora
/:cl:

CakeQ commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -54,3 +54,47 @@
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	signs = list()
```
You're re-defining a lot of variables that are already defined as the same in the parent class. You don't need to re-define variables unless you are changing what they are for the child class

CakeQ commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -54,3 +54,47 @@
+	holocreator_busy = FALSE
+
+/obj/item/holosign_creator/stellascope/use_after(atom/target, mob/living/user, click_parameters)
+	var/turf/T = get_turf(target)
```
keep variable names descriptive (T -> target_turf)

CakeQ commented on code/game/objects/structures/holosigns.dm:
```diff
@@ -50,3 +50,11 @@
+/obj/structure/holosign/constellation
+	name = "starry projection"
+	desc = "A holographic projection of a star system."
+	anchored = TRUE
```
Same as above, you don't need to re-define already defined variables with the same value

CakeQ commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -54,3 +54,40 @@
+/obj/item/holosign_creator/stellascope/use_after(atom/target, mob/living/user, click_parameters)
+	var/turf/target_turf = get_turf(target)
+	if (!target_turf) return FALSE
+	var/obj/structure/holosign/H = locate(holosign_type) in target_turf
```
H -> holosign

XanderDox committed:
Ports/Adds Stellscope from Aurora

Ports the Stellascope from Aurora, reconfiguring it as a holosign creator. Adds it to the loadout for space nerds.

XanderDox commented:
> > Links to the original sourced items. Aurora is under an GNU Affero General Public License v3.0 so this should be fine? I am new to license stuff :)
> 
> Depending on lineage - but generally speaking - there are multiple licenses applied to different parts of any given spaceman source material. This is because in fossville, _code_ licenses and _asset_ licenses have little conceptual overlap (the authors of each were thinking of fairly different things even though they often go together), and it's ended up that the common **distribution** license for media assets from tg and bay derived games (or, the vast majority of them) is CC3-BY-SA, and similarly the code gubbins **distribution** license is AGPL3.
> 
> Because aurora.3 is derived from bay after the adoption of AGPL (which followed TG's move to AGPL for then-compatibility reasons) what this means most of the time is that we can freely use code from same-lineage codebases with git itself as a living credit history. Media assets are a lot muddier because of churn between different codebases, and we want to do our best to identify the real original author and license - not to cover our asses, but just to do right by creators.
> 
> ed: This is why we have a whole set of home grown airlock sounds. The widely used originals were identified as nabbed from a game; when we ID this stuff, we do our best to fix it.

In this case after some digging, which is made hard by the relevant PRs no longer being viewable on github, it seems the original author of the PR that added these things was Alberyk.

HeyBanditoz commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -41,16 +41,53 @@
+	creation_time = 5
+	holosign_type = /obj/structure/holosign/constellation
+
+/obj/item/holosign_creator/stellascope/use_after(atom/target, mob/living/user, click_parameters)
```
you'll want to extract all this logic out to a common proc (and have both `/obj/item/holosign_creator/stellascope/use_after` and `/obj/item/holosign_creator/stellascope` call it), since it's mostly a copypaste between the `use_after` above.

something like this should work:
```ts
/obj/item/holosign_creator/create_hologram(atom/target, mob/living/user, click_parameters)
    // logic here
```
you'll also want a `var/sound` on the holosign creator so you can use the jingle on the stellascope, instead

and, in the new proc, style nit, but convert all `if(thing)` to `if (thing)` and `for(thing)` to `for (thing)`

HeyBanditoz commented on icons/obj/toy.dmi:
did you mean to remove this file? looks pretty busy:	
<img width="242" height="193" alt="VSCodium_KojmTz16i7" src="https://github.com/user-attachments/assets/a7184bee-61c9-4b79-b83a-67d02bd679ba" />

HeyBanditoz commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -41,16 +42,32 @@
+	create_sound = 'sound/effects/jingle.ogg'
+
+/obj/item/holosign_creator/stellascope/use_after(atom/target, mob/living/user, click_parameters)
+	create_holosign(target, user, click_parameters)
```
you can remove this, since the base proc (on line 53) is doing it already!
```suggestion
```

XanderDox replied:
Removed :)

SierraKomodo commented on code/game/objects/structures/holosigns.dm:
```diff
@@ -50,3 +50,10 @@
+/obj/structure/holosign/constellation
+	name = "starry projection"
+	desc = "A holographic projection of a star system."
+	icon = 'icons/obj/hologram_projections.dmi'
```
Identical to parent.

```suggestion
```

XanderDox replied:
oops, holdover from when they were in separate icon files.
